build-docker/action.yaml

```diff
@@ -0,0 +1,101 @@
+    default: .docker-config.json
+  dockerhub-user:
+    required: true
+    description: username for dockerhub
+  dockerhub-password:
+    required: true
+    description: password for dockerhub
+  github-token:
+    required: true
+    description: Usually secrets.GITHUB_TOKEN
+  npm-auth-token:
+    required: false
+    description: The Node Package Manager (npm) authentication token. This token is used to authenticate against a private NPM registry configured via a .npmrc file. Gets pass to the docker build as a secret
+  npm-token:
+    required: false
+    description: The Node Package Manager (npm) authentication token. This token is used to authenticate against the NPM registry. Gets passed to the docker build as a secret
+  image-version:
+    required: true
+    description: Docker image version
+  image-platform:
+    description: Target platform to build image for (eg. linux/amd64 (default), linux/arm64, etc)
+    required: false
+    default: linux/amd64
+  docker-metadata-tags:
+    description: "'List of tags as key-value pair attributes' See: https://github.com/docker/metadata-action#tags-input"
+    required: false
+outputs:
+  image-name:
+    description: Docker image name
+    value: ${{ steps.config.outputs.image-name }}
+  image-with-tag:
+    description: Full image with tag - <image-name>:<image-version>
+    value: ${{ steps.config.outputs.image-name }}:${{ inputs.image-version }}
+
+runs:
+  using: composite
+  steps:
+    - name: Dump GitHub context
+      shell: bash
+      env:
+        GITHUB_CONTEXT: ${{ toJson(github) }}
+      run: echo "$GITHUB_CONTEXT"
+    - id: config
+      shell: bash
+      run: |
+        # if docker-config file does not exist - fail
+        if [ ! -f "${{ inputs.docker-config-file }}" ]; then
+          echo "::error::Docker config file not found: ${{ inputs.docker-config-file }}" && exit 1
+        fi
+        image_name=$(jq -r .imageName ${{ inputs.docker-config-file }})
+        echo "image-name: ${image_name}"
+        echo "image-name=${image_name}" >> $GITHUB_OUTPUT
+        dockerfile=$(jq -r '.dockerfile // "./Dockerfile"' ${{ inputs.docker-config-file }})
+        echo "Dockerfile: ${dockerfile}"
+        echo "dockerfile=${dockerfile}" >> $GITHUB_OUTPUT
+        echo "image-version: ${{ inputs.image-version }}"
+        echo "image-platform: ${{ inputs.image-platform }}"
+        echo "docker-metadata-tags: ${{ inputs.docker-metadata-tags }}"
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ inputs.dockerhub-user }}
+        password: ${{ inputs.dockerhub-password }}
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: |
+          ${{ steps.config.outputs.image-name }}
+        flavor: |
```

Comment:
Can we also support customizing flavor in the docker-config file? We have an use case for it in @turo/front-end releases 💡

Reply:
Added!